src/omit.ts

```diff
@@ -28,22 +29,37 @@ export function omit<T extends Record<PropertyKey, any>, K extends keyof T>(
  * @category Object
  */
 export function omit<K extends PropertyKey>(
-  names: ReadonlyArray<K>
-): <T extends Record<PropertyKey, any>>(object: T) => Omit<T, K>;
+  propNames: ReadonlyArray<K>
+): <T extends object>(data: T) => Omit<T, K>;
 
 export function omit() {
   return purry(_omit, arguments);
 }
 
-function _omit<T extends Record<PropertyKey, any>, K extends keyof T>(
-  object: T,
-  names: Array<K>
+function _omit<T extends object, K extends keyof T>(
+  data: T,
+  propNames: ReadonlyArray<K>
 ): Omit<T, K> {
-  const set = new Set(names as Array<string>);
-  return Object.entries(object).reduce<any>((acc, [name, value]) => {
-    if (!set.has(name)) {
-      acc[name] = value;
-    }
-    return acc;
-  }, {}) as Omit<T, K>;
+  if (propNames.length === 0) {
+    return { ...data };
+  }
```

Comment:
could we make this case impossible on type level? What's the point of calling omit with an empty array of propNames? Making it a `NonEmptyArray` on type level would rid us of that path.

Reply:
I thought about it too. My thinking here was that someone might want to dynamically omit fields, maybe based on some UI component that defines the omitted parts, and that might contain a trivial empty case. Or if you want to optionally omit something and don't want to do:
```ts
const x = pipe(
  input,
  ...,
  omit(<CONDITION> ? [] : ["prop"]),
);

// Instead of 
const y = pipe(
  input,
  ...,
  <CONDITION> ? identity : omit(["prop"]),
  ...,
);

Reply:
I would argue "don't call `omit` if you don't want to omit anything" ? In a pipe, the second solution looks fine. It conceptually quite similar to why we don't allow null/undefined to be passed to array methods - the check needs to be explicitly moved to the outside, _before_ the invocation of e.g. `filter`

Reply:
OK, but that creates an issue with how we change the signature of `omit` now that it's in the wild without breaking people's code :) ?